GPU/GPUTracking/Merger/GPUTPCGMTrackParam.cxx

```diff
@@ -494,16 +495,21 @@ GPUd() void GPUTPCGMTrackParam::AttachClusters(const GPUTPCGMMerger* GPUrestrict
   const float z0 = row.Grid().ZMin() - zOffset; // We can use our own ZOffset, since this is only used temporarily anyway
   const float stepZ = row.HstepZ();
   int bin, ny, nz;
-  const float tube = 2.5f;
+
+  float err2Y, err2Z;
+  Merger->Param().GetClusterErrors2(iRow, Z, mP[2], mP[3], err2Y, err2Z);
+  const float sy2 = CAMath::Min(Merger->Param().rec.tpcTubeMaxSize2, Merger->Param().rec.tpcTubeChi2 * (err2Y + mC[0]));
```

Comment:
This is a quite simple improved check for the TPC tube, using the track cov and the cluster error estimation, with a scaler factor and a max limit to avoid crazy cov entries. It already give much better results, 4% less clusters become fake protected, while almost no correctly protected clusters become unprotected.
Parameters should be tuned, one could also think about a combined check for y and z instead of applying them independently.
Also, the criterion is currently not applied correctly for the looper interpolation between the legs.